### DISCUSSION TITLE
Do Python formatters respect pyproject.toml?

### DISCUSSION THREAD

**Discussion by declension** [numtide/treefmt#283] [Help]
Ruff and Mypy are two of the biggest, but I can't convince `treefmt` (via `treefmt-nix` at least) to respect settings included in `pyproject.toml` which is where most modern tooling is now providing config space in.

Is this supported?
Thanks!

**Comment by brianmcgee** [comment]
Sorry for the late reply.

Do you have any examples I could look at?

**Reply by jfly** [reply]
Here's a simple demo of ruff settings in `pyproject.toml` being ignored by `treefmt-nix`: <https://github.com/jfly/2025-01-16-treefmt-pyproject-demo>. I haven't dug into why. Is it possible that we're invoking ruff in a temp dir where pyproject.toml doesn't exist?
